transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java

```diff
@@ -373,6 +374,18 @@ protected boolean login(NextFilter nextFilter,
                     }
                     // Login was successful, but the subject identified does not have required roles
                     // to create the web socket.  We re-issue a challenge here.
+                    if (realmIndex + 1 < realms.length && resultType == SUCCESS) {
+                        // we only enforce subject at the end of the realm chain, otherwise we skip to the next realm
+                        loginContexts[realmIndex] = loginContext;
+
+                        // remember login context
+                        httpRequest.setLoginContext(loginContext);
+
+                        // remember subject
+                        httpRequest.setSubject((loginContext == null || loginContext == LOGIN_CONTEXT_OK) ? subject : loginContext.getSubject());
+//                        realmIndex++;
+                        return true;
+                    }
```

Comment:
Does `resultType` `SUCCESS` mean that all factors have succeeded or just the current factor?

If it means only the current factor has succeeded, then this seems to be returning `true` eagerly, even before all factors have been processed, right?